services/api/mw2pdf/src/classes/MediaWikiSession.ts

```diff
@@ -1,131 +1,219 @@
     if (this.isAuthenticated(domain)) {
       return true
     }
 
     // Step 1: Generate a login token
-    const loginTokenRequest = await makeLoginTokenRequest(apiUrl)
-    const authCookies = parseCookies(domain, loginTokenRequest)
-    const loginToken = await extractLoginToken(loginTokenRequest)
+    const loginTokenRequest = await makeLoginTokenRequest(apiUrl);
+    const authCookies = parseCookies(domain, loginTokenRequest);
+    const loginToken = await extractLoginToken(loginTokenRequest);
 
     // Step 2: Perform the login
-    const loginRequest = await makeLoginRequest({
+    const loginRequest = await makeLoginRequest(
       apiUrl,
       username,
       password,
       loginToken,
-      cookies: authCookies,
-    })
+      authCookies,
+    )
+
+    // Step 3: Set the cookies for the session
     this.setCookies(
       domain,
       parseCookies(domain, loginRequest),
     )
-    return true
+
+    return true;
   }
 
-  async makePdf(url, tmpDirectory = './', includeTitlePage = false) {
-    const pdfGenerator = new PdfGenerator(tmpDirectory)
+  async makePdf(url: URL, options: MwPdfOptions): Promise<Pdf> {
+    // Extract the domain (as _domain_) from the url.
+    const domain = url.host;
+
+    /**
+     * Begin puppeteer
+     */
+
+    // Open the browser
     const browser = await puppeteer.launch(
       { args: ['--no-sandbox', '--disable-setuid-sandbox'] }
-    )
-    const page = await browser.newPage()
-    const { host: domain } = new URL(url)
-    await page.setCookie(...this.getCookies(domain))
-    await page.goto(url, { waitUntil: 'networkidle2' })
+    );
+
+    // Get a new page
+    const page = await browser.newPage();
+
+    // Set cookies into the browser that we pull from the MW session. This means
+    // that all the login and authentication we did before will be valid.
+    await page.setCookie(...this.getCookies(domain));
+
+    // Now, go to the page!
+    await page.goto(url.toString(), { waitUntil: 'networkidle2' });
+
+    // Get the page's title.
     const pageTitle = await page.title()
-    const pagePdf = pdfGenerator
-      .generatePdfObject({ title: pageTitle })
-    await page.pdf({ path: pagePdf.path, format: 'A4' })
-    await browser.close()
-    if (includeTitlePage === true) {
-      const titlePdf = await pdfGenerator
-        .generateTitlePagePdf(pageTitle)
-      const mergedPdf = await pdfGenerator
-        .generateMergedPdf(
-          [titlePdf, pagePdf],
-          pdfGenerator.generatePdfObject({ title: pageTitle }),
-        )
-      await deletePdfs([titlePdf, pagePdf])
-      return mergedPdf
+    const pageFilename = join(options.workDirectory, `${v4()}.pdf`);
+
+    // Generate a Pdf object with the appropriate title.
+    const pagePdf: Pdf = PdfFactory.generatePdfObject(new PdfConstructorOptions(pageTitle, pageFilename));
+
+    // print the visible page into a PDF
+    await page.pdf({ path: pagePdf.filename, format: 'A4' });
+
+    // close the browser
```

Comment:
I don't think we need this comment since it's just a repeat of the below line (`browser.close` vs `close the browser`)

Reply:
As if it never appeared :-)